### PR TITLE
[Proposal] Add turbulance disturbance to LunarLander

### PR DESCRIPTION
**Pull request by virgilt**
# Description

Added a torque function for added turbulence to the lander. The torque follows the same magnitude as the linear wind function.

Fixes # (2799)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by virgilt** [comment]
There is a mistake in this pull request that I need to fix.

**Comment by agentydragon** [comment]
For this kind of change, you should increase the version number of the environment. And probably would also be better to make the turbulence optional.

**Comment by pseudo-rnd-thoughts** [comment]
This functionality I believe is turned off by default so it shouldn't affect any previous code but you are probably right that increasing the version number would be helpful to log the change in environment functionality.

**Comment by virgilt** [comment]
@jjshoots I ran the provided tests and the lander is working. Also reran `pre-commit run --all-files`.

**Comment by virgilt** [comment]
Here is the test that I ran:
```

import gym
env = gym.make("LunarLander-v2", enable_wind=True)
observation, info = env.reset(seed=42, return_info=True)
for _ in range(1000):
   env.render()
   action = env.action_space.sample()
   observation, reward, done, info = env.step(action)

   if done:
      observation, info = env.reset(return_info=True)
env.close()
```

**Comment by jjshoots** [comment]
> This functionality I believe is turned off by default so it shouldn't affect any previous code but you are probably right that increasing the version number would be helpful to log the change in environment functionality.

This is just my opinion but I don't think incrementing version numbers is necessary here right? If it doesn't change/break the functionality of the current code and still works with everyone else's use of LunarLander-v2, incrementing version numbers doesn't really do anything for the project imho (and may cause more people's implementations to break because of version number changes.

**Comment by pseudo-rnd-thoughts** [comment]
@virgilt At the very least, could you update the version history in the documentation with the change in a similar way to car racing

**Comment by pseudo-rnd-thoughts** [comment]
@virgilt a gif would be great just to see what environment looks like with these changes

**Comment by pseudo-rnd-thoughts** [comment]
Sorry, in addition, could you add a test that this feature works. @jjshoots any good ideas of testing functionality other than, with different wind powers, the observations are different. 
@virgilt Could this push the robot out of the observation space? If so, could you clip the wind to the observation space or abapt the observation space to the wind power.

**Comment by jjshoots** [comment]
@pseudo-rnd-thoughts The observations are actually the same with and without wind, since wind is an unobservable thing in the env (making the env a POMDP instead of just an MDP as it was previously).

The wind can push the lander outside of the space, yes, but that also means the environment ends. Right now we're testing to make sure that the wind doesn't result in an unlearnable env because of the wind being more powerful than what the motors can meaningfully reject.

**Comment by virgilt** [comment]
@pseudo-rnd-thoughts I added version history within lunar_lander.py ...Do I still need to add a gif, or is that no longer needed because of @jjshoots explanation?

@jjshoots I added torque_power to kwargs and an assert.

**Comment by pseudo-rnd-thoughts** [comment]
Ok, we cannot see this effect then don't worry about the gif @virgilt.

**Comment by virgilt** [comment]
@jjshoots what do you think about removing the upper bound on wind_power and torque_power, and just putting a recommended range within the docs?

**Comment by jjshoots** [comment]
@virgilt This is fair too, but I think @pseudo-rnd-thoughts is a better candidate to handle this question.

The context is pretty much that wind and turbulence need to be within a particular range for the environment to be solvable. Right now it's handled with an assert that keeps the value in that range, but Travis is suggesting that we remove the assert and put the recommended range in the docs.

**Comment by jjshoots** [comment]
Here's a gif of a somewhat trained agent in the environment with wind enabled. In some instances the lander actually crashes, but I think this is something that can be improved with more learning/tweaking hyperparams.

Peek 2022-05-13 21-10

**Comment by jjshoots** [comment]
@virgilt can you change `torque_power` to perhaps `turbulence_power`, this makes things slightly more descriptive. Other than that, LGTM!

**Comment by virgilt** [comment]
@jjshoots I change the variable name in all places.

**Comment by virgilt** [comment]
> @virgilt This is fair too, but I think @pseudo-rnd-thoughts is a better candidate to handle this question.
> 
> The context is pretty much that wind and turbulence need to be within a particular range for the environment to be solvable. Right now it's handled with an assert that keeps the value in that range, but Travis is suggesting that we remove the assert and put the recommended range in the docs.

@jjshoots, these changes were suggested by @pseudo-rnd-thoughts. We wanted your opinion on it.

**Comment by jjshoots** [comment]
If that's the case, then I'm voting for removing the asserts, changing them to warnings as well as putting it in the docs :)

**Comment by pseudo-rnd-thoughts** [comment]
In addition, could you update the environment docstring
Currently, it is "`wind_power` dictates the maximum magnitude of wind."
Could you it talk about torque power and the recommended range of both parameters

**Comment by virgilt** [comment]
I’ll try to have to this wrapped up either tonight or tomorrow.

**Comment by virgilt** [comment]
@jjshoots @pseudo-rnd-thoughts am I only changing the asserts for wind_power and turbulence_power? Or am I also changing gravity?

**Comment by virgilt** [comment]
@jjshoots @pseudo-rnd-thoughts I added warnings, removed assert, and updated docs with definition/recommendations for wind_power and turbulence_power.

File: gym/envs/box2d/lunar_lander.py [hunk 1]
```diff
@@ -153,10 +155,11 @@ class LunarLander(gym.Env, EzPickle):
     `k` is set to 0.01.
     `C` is sampled randomly between -9999 and 9999.
 
-    `wind_power` dictates the maximum magnitude of wind.
+    `wind_power` dictates the maximum magnitude of linear wind applied to the craft. The recommended value for `wind_power` is between 0.0 and 20.0.
+    `turbulence_power` dictates the maximum magnitude of rotational wind applied to the craft. The recommended value for `turbulence_power` is between 0.0 and 2.0.
 
     ### Version History
-    - v2: Count energy spent
+    - v2: Count energy spent and in 0.24, added turbulance with wind power and turbulence_power parameters
```
Comment (jjshoots) [inline gym/envs/box2d/lunar_lander.py:162]: This is a minor nitpick, but what is `and in 0.24`?

**Comment by jjshoots** [comment]
@pseudo-rnd-thoughts Other than the minor question on the version documentation, LGTM.

File: gym/envs/box2d/lunar_lander.py (hunk 1 above)
Comment (virgilt) [reply gym/envs/box2d/lunar_lander.py:162]: It's the gym version. Should I change it to v0.24 for less confusion?

Comment (pseudo-rnd-thoughts) [reply gym/envs/box2d/lunar_lander.py:162]: To make it easiest to understand, yes, but no critical

**Pull request merged**